google/cloud/sql/connector/utils.py

```diff
@@ -57,6 +57,15 @@ async def generate_keys():
     return priv_key, pub_key
 
 
+async def await_task_cancellation(task: asyncio.Task) -> None:
+    """A helper coroutine to await the cancellation of a task."""
+    task.cancel()
+    try:
+        await task
```

Comment:
Just to confirm -  are we sure we need to actually await the results? What happens if we just cancel?

Reply:
awaiting the task guarantees that it was actually cancelled, since tasks can swallow the `CancelledError` and continue to run. https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancel

Reply:
It doesn't guarantee it was cancelled, just that the task was resolved:
 
> This arranges for a CancelledError exception to be thrown into the wrapped coroutine on the next cycle of the event loop.

This means that the CancelledError will get potentially thrown any time there is an "await" statement

> Therefore, unlike Future.cancel(), Task.cancel() does not guarantee that the Task will be cancelled, although suppressing cancellation completely is not common and is actively discouraged.

So you can cancel a task, but if if captures the error and suppresses, it just won't be canceled. 

So I don't think we need to await the result, because we're only executing one coro at a time (which means the task will be canceled on next context switch to it) and none of the functions we are calling are capturing the interruptions. If they were, we don't really care because we don't need the result anyway.